Mathlib/CategoryTheory/Distributive/Monoidal.lean

```diff
@@ -0,0 +1,298 @@
+    PreservesColimit (pair Y Z) (tensorRight X) :=
+  (IsMonoidalRightDistrib.preservesBinaryCoproducts_tensorRight X).preservesColimit
+
+/-- The canonical right distributivity isomorphism -/
+def rightDistrib [IsMonoidalRightDistrib C] (X Y Z : C) : (Y ⊗ X) ⨿ (Z ⊗ X) ≅ (Y ⨿ Z) ⊗ X :=
+  PreservesColimitPair.iso (tensorRight X) Y Z
+
+end IsMonoidalRightDistrib
+
+namespace Distributive
+
+/-- Notation for the forward direction morphism of the canonical right distributivity isomorphism -/
+notation "∂R" => rightDistrib
+
+end Distributive
+
+open Distributive
+
+instance IsMonoidalRightDistrib.isIso_rightDistrib_hom [IsMonoidalRightDistrib C] {X Y Z : C} :
+    IsIso (∂R X Y Z).hom :=
+  isIso_hom <| rightDistrib X Y Z
+
+instance IsMonoidalRightDistrib.of_isIso_coprodComparisonTensorRight
+    [i : ∀ {X Y Z : C}, IsIso (coprodComparison (tensorRight X) Y Z)] :
+    IsMonoidalRightDistrib C where
+  preservesBinaryCoproducts_tensorRight _ :=
+    ⟨preservesBinaryCoproducts_of_isIso_coprodComparison _ |>.preservesColimit⟩
+
+/-- The forward direction of the right distributivity isomorphism is equal to the cogap morphism
+`coprod.desc (coprod.inl ▷ _) (coprod.inr ▷ _) : (Y ⊗ X) ⨿ (Z ⊗ X) ⟶ (Y ⨿ Z) ⊗ X`. -/
+lemma rightDistrib_hom [IsMonoidalRightDistrib C] {X Y Z : C} :
+  (∂R X Y Z).hom = coprod.desc (coprod.inl ▷ _) (coprod.inr ▷ _) := by rfl
+
+@[reassoc (attr := simp)]
+lemma coprod_inl_rightDistrib_hom [IsMonoidalRightDistrib C] {X Y Z : C} :
+    coprod.inl ≫ (∂R X Y Z).hom = coprod.inl ▷ X := by
+  rw [rightDistrib_hom, coprod.inl_desc]
+
+@[reassoc (attr := simp)]
+lemma coprod_inr_rightDistrib_hom [IsMonoidalRightDistrib C] {X Y Z : C} :
+    coprod.inr ≫ (∂R X Y Z).hom = coprod.inr ▷ X := by
+  rw [rightDistrib_hom, coprod.inr_desc]
+
+/-- The composite of `(coprod.inl ▷ X) : Y ⊗ X ⟶ (Y ⨿ Z) ⊗ X` and
+`(∂R X Y Z).inv :  (Y ⨿ Z) ⊗ X ⟶ (Y ⊗ X) ⨿ (Z ⊗ X)` is equal to the left coprojection
+`coprod.inl : Y ⊗ X ⟶ (Y ⊗ X) ⨿ (Z ⊗ X)`. -/
+@[reassoc (attr := simp)]
+lemma whiskerRight_coprod_inl_rightDistrib_inv [IsMonoidalRightDistrib C] {X Y Z : C} :
+    (coprod.inl ▷ X) ≫ (∂R X Y Z).inv = coprod.inl := by
+  apply (cancel_iso_hom_right _ _ (∂R X Y Z)).mp
+  rw [assoc, Iso.inv_hom_id, comp_id, coprod_inl_rightDistrib_hom]
+
+/-- The composite of `(coprod.inr ▷ X) : Z ⊗ X ⟶ (Y ⨿ Z) ⊗ X` and
+`(∂R X Y Z).inv :  (Y ⨿ Z) ⊗ X ⟶ (Y ⊗ X) ⨿ (Z ⊗ X)` is equal to the right coprojection
+`coprod.inr : Z ⊗ X ⟶ (Y ⊗ X) ⨿ (Z ⊗ X)`. -/
+@[reassoc (attr := simp)]
+lemma whiskerRight_coprod_inr_rightDistrib_inv [IsMonoidalRightDistrib C] {X Y Z : C} :
+    (coprod.inr ▷ X) ≫ (∂R X Y Z).inv = coprod.inr := by
+  apply (cancel_iso_hom_right _ _ (∂R X Y Z)).mp
+  rw [assoc, Iso.inv_hom_id, comp_id, coprod_inr_rightDistrib_hom]
+
+/-- In a symmetric monoidal category, the left distributivity is equal to
+the right distributivity up to braiding isomorphisms. -/
+@[simp]
+lemma SymmetricCategory.leftDistrib_braiding [BraidedCategory C] {X Y Z : C} :
```

Comment:
The name for this lemma should be `coprodComparison_tensorLeft_braiding_hom` (similarly below).